### PR TITLE
Playwright test for user creation inside department

### LINKED ISSUE #14428: Playwright test for user creation
**Describe the bug**
Write Playwright test for user creation

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

---

### Instructions for Requesting Assignment:

To request assignment, please clearly outline your solution and timeline by commenting on the issue using the format below:

**Describe your solution clearly:**
Provide a detailed explanation of your proposed solution, including your approach, key implementation steps, and relevant examples or references. Mention any dependencies, assumptions, or risks you foresee that might affect your timeline or implementation.

**Expected Timeline:**
- End date: [Expected submission date of a completed Pull Request]

**Additional Context:**
Include any other relevant context, links, screenshots, or resources that support your proposed solution.

> 🚨 Your assignment may be unassigned if there is no activity or progress within the stated timeline unless communicated clearly and agreed upon.

### PR DESCRIPTION
**Pull request by NikhilA8606**
- Fix : #14428 
- Wrote playwright test for user creation

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is placed in I18n files.
- [ ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [ ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.
- [x] Add or update Playwright tests for related changes

## Summary by CodeRabbit

* **Tests**
  * Added end-to-end tests for user creation and department linkage: covers form entry with randomized data, role selection, linking to an organization/department, success notifications, search verification that the new user appears, and validation of default form values for user type and password setup to improve reliability of administrative workflows.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by nihal467 — CHANGES_REQUESTED** [review]
remove data-cy
follow standardization of existing test files
migrating to playwright standards
modulize the test
add additional test for negative error validation
users faker array wherever necessary
remove consoles

**Comment by nihal467** [comment]
- change the PR to follow the .step approach
- its a user creation inside a department so change the file placing

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+test.use({ storageState: "tests/.auth/user.json" });
+
+test.describe("User Management in Departments", () => {
+  let facilityId: string;
+
+  test.beforeEach(async ({ page }) => {
+    facilityId = getFacilityId();
+
+    await page.goto(`/facility/${facilityId}/settings/departments`);
+    await page.getByRole("table").getByRole("row").nth(1).click();
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: - select admini department always

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    await test.step("Navigate to user creation form", async () => {
+      await page.getByRole("button", { name: "Add User" }).click();
+      await expect(
+        page.getByRole("heading", { name: "Add New User" }),
+      ).toBeVisible();
+    });
+
+    await test.step("Fill and submit user creation form", async () => {
+      await page.getByRole("combobox", { name: "User Type *" }).click();
+      await page.getByRole("option").first().click();
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: - make this a faker array

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+      await page
+        .getByRole("textbox", { name: "Password", exact: true })
+        .fill(password);
+      await page
+        .getByRole("textbox", { name: "Confirm Password *" })
+        .fill(password);
+      await page
+        .getByRole("textbox", { name: "Phone Number *" })
+        .fill(phoneNumber);
+      await page.getByRole("combobox", { name: "Gender *" }).click();
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: - make it a faker array

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+      ]).catch(() => {});
+
+      const isLinkUserVisible = await linkUserHeading
+        .isVisible()
+        .catch(() => false);
+      const isRoleVisible = await roleCombobox.isVisible().catch(() => false);
+
+      if (isLinkUserVisible || isRoleVisible) {
+        await roleCombobox.click();
+        await page.getByRole("option").first().click();
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: - use faker

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+        .last()
+        .locator("..")
+        .locator("..");
+
+      await expect(lastUserCard).toContainText(firstName, { timeout: 10000 });
+      await expect(lastUserCard).toContainText(lastName, { timeout: 10000 });
+    });
+  });
+
+  test("Open user creation form with default values", async ({ page }) => {
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: - why do we even need this test ?

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+    });
+
+    await test.step("Verify form submission is prevented", async () => {
+      await page.getByRole("button", { name: "Create User" }).click();
+      await page.waitForTimeout(1000);
+
+      const dialogStillVisible = await page
+        .getByRole("dialog", { name: "Add New User" })
+        .isVisible();
+      expect(dialogStillVisible).toBe(true);
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: - verify the error message creation is rejected

File: tests/facility/settings/departments/userManagement.spec.ts [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+      await page
+        .getByRole("textbox", { name: "Password", exact: true })
+        .fill(password);
+      await page
+        .getByRole("textbox", { name: "Confirm Password *" })
+        .fill("DifferentPassword@123");
+    });
+
+    await test.step("Verify password mismatch error is shown", async () => {
+      await page.waitForTimeout(500);
```
Comment (nihal467) [inline tests/facility/settings/departments/userManagement.spec.ts]: look into copilot thing

**Review by nihal467 — CHANGES_REQUESTED** [review]

**Review by nihal467 — APPROVED** [review]
